src/gas-api.js

```diff
@@ -1,210 +1,207 @@
-      if (!response.startsWith('//OK')) reject(new Error('Update file failed'));
-      resolve();
+      if (!response.startsWith('//OK')) throw new Error('Init failed');
+      //evil eval, but it's simple to get the object since it's not valid json object
+      const initData = eval(response.slice(4)).filter((e) => {
+        return typeof(e) === 'object';
+      })[0];
+      const ids = initData.filter((data) => { return /^\w{8}-\w{4}-\w{4}-\w{4}-\w{12}$/.test(data) });
+      context.projectId = initData[initData.indexOf(ids[0]) + 1];
+      if (context.projectId.length != 33) {
+        reject(new Error('cant not get project ID'));
+      }
+      const promises = ids.map((id) => {
+        return new Promise((resolve, reject) => {
+          const payload = `7|1|8|${getBaseUrl()}\|${context.gasToken}|_|getFileContent|k|${id}|${context.projectId}|l|1|2|3|4|1|5|5|6|7|8|0|0|`;
+          $.ajax({
+            url: context.gasUrl,
+            headers: context.gasHeaders,
+            method: 'POST',
+            crossDomain: true,
+            data: payload,
+            dataType: 'text'
+          })
+          .then((response) => {
+            if (!response.startsWith('//OK'))  reject(new Error('get apps script code failed'));
+            //evil eval, but it's simple to get the object since it's not valid json object
+            const codeContent = eval(response.slice(4)).filter((e) => {
+              return typeof(e) === 'object';
+            })[0];
+            resolve({file : codeContent[codeContent.length - 6], content: codeContent[codeContent.length - 10], id : id });
+          })
+          .fail(reject);
+        })
+      });
+      return Promise.all(promises);
     })
-    .fail((err) => {
-      reject(new Error('Update file failed'));
-    });
-  });
-}
+    .then((responses) => {
+      context.fileIds = responses.reduce((hash, elem) => {
+        if (elem) hash[elem.file] = elem.id;
+        return hash;
+      }, {});
+      return responses;
+    })
+  }
 
-function gasDeleteFile(file) {
-  const payload = `7|1|4|${getBaseUrl()}\|${context.gasToken}|_|deleteFile|1|2|3|4|0|`;
-  let headers = context.gasHeaders;
-  Object.assign(headers, { 'file-id': context.fileIds[file]});
-  return new Promise((resolve, reject) => {
-    $.ajax({
-      url: context.gasUrl,
-      headers: headers,
-      method: 'POST',
-      crossDomain: true,
-      data: payload,
-      dataType: 'text'
+  gasCreateFile(file, type) {
+    const typeId = type === 'gs' ? 0 : 2;
+    const payload = `7|1|7|${getBaseUrl()}\|${context.gasToken}|_|makeNewFile|19|h|${file}|1|2|3|4|2|5|6|7|6|${typeId}|`;
+    return new Promise((resolve, reject) => {
+      $.ajax({
+        url: context.gasUrl,
+        headers: context.gasHeaders,
+        method: 'POST',
+        crossDomain: true,
+        data: payload,
+        dataType: 'text'
+      })
+      .then(resolve)
+      .fail((err) => {reject(new Error('Create file failed'))})
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/76f8108a25ffe1858d5d27c3d7dabf83cc8b0291/src/gas-api.js#L144"></span>‘err’ is defined but never used.</td>
    </tr>
  </tbody>
</table>